dot/parachain/bitfield-distribution/bitfield_distribution.go

```diff
@@ -235,19 +249,79 @@ func (b *BitfieldDistribution) processPeerConnectedEvent(event networkbridgeeven
 }
 
 func (b *BitfieldDistribution) processPeerDisconnectedEvent(event networkbridgeevents.PeerDisconnected) {
+	logger.Infof("peer disconnected: %s", event.PeerID)
+
 	b.mu.Lock()
 	delete(b.peerViews, event.PeerID)
 	b.mu.Unlock()
 }
 
 func (b *BitfieldDistribution) processNewGossipTopologyEvent(event networkbridgeevents.NewGossipTopology) error {
-	//TODO implement in #4357
-	panic("implement me")
+	logger.Infof("process NewGossipTopology event")
+
+	sessionIdx := event.Session
+	newTopology := event.Topotogy
+	prevNeighbors := b.topologies.CurrentTopology.LocalNeighbours
+
+	peers := make(map[peer.ID]struct{})
+	for _, val := range newTopology.PeerIDs {
+		peers[val] = struct{}{}
+	}
+
+	shuffledIndices := make([]uint, len(event.Topotogy.ShuffledIndices))
+	for i, v := range event.Topotogy.ShuffledIndices {
+		shuffledIndices[i] = uint(v)
+	}
+
+	canonicalShuffling := make([]grid.TopologyPeerInfo, 0)
+	for i, info := range event.Topotogy.CanonicalShuffling {
+		t := grid.TopologyPeerInfo{
+			Peers:          info.PeerID,
+			ValidatorIndex: info.ValidatorIndex,
+			DiscoveryID:    types.AuthorityID(info.DiscoveryID),
+		}
+		canonicalShuffling[i] = t
+	}
```

Comment:
need to preallocate here otherwise the `canonicalShuffling[i] = t` will panic
```suggestion
	canonicalShuffling := make([]grid.TopologyPeerInfo, len(event.Topotogy.CanonicalShuffling))
```